drivers/cyw43/cyw43_ll.h

```diff
@@ -57,6 +57,14 @@ enum {
     CYW43_ITF_AP,
 };
 
+enum {
+    CYW43_AUTH_OPEN,
+    CYW43_AUTH_WEP,
+    CYW43_AUTH_WPA_PSK,
+    CYW43_AUTH_WPA2_PSK,
+    CYW43_AUTH_WPA_WPA2_PSK,
```

Comment:
I don't think these constants will map correctly to the internal AUTH settings in the CYW43.

Reply:
indeed it doesn't.

I did a little reverse engineering on this and this is the result

`
// auth mode configuration
// coded on 3 bits
// [AES][TKIP][802.11G]

#define CYW43_802_11G                  (0x1) // !802_11G == 802_11N
#define CYW43_TKIP                     (0x2)
#define CYW43_AES                      (0x4)
#define CYW43_AUTH_OPEN                        (0x0) // 72.2Mbps OPEN 802.11n 20Mhz
#define CYW43_AUTH_WPA                 (CYW43_802_11G) // 54Mbps PSK-CCMP 802.11g 20Mhz
#define CYW43_AUTH_WPA_TKIP            (CYW43_TKIP) // 54Mbps PSK-TKIP 802.11n 20Mhz
#define CYW43_AUTH_WPA_TKIP_802_11G    (CYW43_TKIP|CYW43_802_11G) // 54Mbps PSK-TKIP 802.11g 20Mhz
#define CYW43_AUTH_WPA_AES             (CYW43_AES) // 72.2Mbps PSK-CCMP 802.11n 20Mhz
#define CYW43_AUTH_WPA_AES_802_11G     (CYW43_AES|CYW43_802_11G) // 54Mbps  PSK-CCMP 802.11g 20Mhz
#define CYW43_AUTH_WPA_AES_TKIP                (CYW43_AES|CYW43_TKIP) // 72.2Mbps PSK-TKIP-CCMP 802.11n 20Mhz
#define CYW43_AUTH_WPA_AES_TKIP_802_11G        (CYW43_AES|CYW43_TKIP|CYW43_802_11G) // 54Mbps PSK-TKIP-CCMP 802.11g 20Mhz

`

@dpgeorge nevertheless we can't set up a wifi AP other than OPEN for the moment, which is not the best default conf, do you think we could at least have some of this patch included in one of the next release ?

Reply:
What's the most trivial change we could make here?